library/types/test/hostname_test.rb

```diff
@@ -10,30 +10,49 @@
 
   let(:etc_hostname) { "etc.hostname.cz" }
   let(:cmd_hostname) { "cmd.hostname.cz" }
+  let(:hostname)     { Yast::Hostname }
+
+  def allow_execute_hostname(result)
+    ret = { "stdout" => result, "exit" => 0 } if result
```

Comment:
there is ```nil``` bcs when it is not valid value I don't care. But, you're right i should use one of 'valid'  return values for invalid case.